inference-engine/tests/functional/inference_engine/caching_test.cpp

```diff
@@ -128,10 +129,10 @@ class MockExecutableNetwork : public ExecutableNetworkInternal {
     MOCK_METHOD1(setNetworkInputs, void(const InputsDataMap networkInputs));
     MOCK_METHOD1(setNetworkOutputs, void(const OutputsDataMap networkOutputs));
 
-    void Export(std::ostream& networkModel) override {
-        std::lock_guard<std::mutex> guard(m_pluginMutex);
-        ExecutableNetworkInternal::Export(networkModel);
-    }
+    // void Export(std::ostream& networkModel) override {
+    //     std::lock_guard<std::mutex> guard(m_pluginMutex);
+    //     ExecutableNetworkInternal::Export(networkModel);
+    // }
```

Comment:
Maybe we should also introduce 'virtual GetPointerToPlugin' and use this in Multidevice plugin instead of direct '_plugin'. And here it will be possible to create a fully thread-safe mock network implementation.
But real chances for race condition here is very-very low and will impact only CI, so maybe it can be done as separate minor PR

Reply:
don't get your point, sorry

Reply:
In ie_executable_network_internal.hpp
`
virtual InferencePlugin::Ptr GetPointerToPlugin() const { return _plugin; }
`
In caching test:
`
InferencePlugin::Ptr GetPointerToPlugin() const override {
  std:lock_guard<...> guard(mutex);
  return ExecutableNetworkInternal::GetPointerToPlugin();
`

In MultiDeviceExecutableNetwork::SetConfig use "GetPointerToPlugin" instead of "this->_plugin"